packages/expo-router/src/useScreens.tsx

```diff
@@ -219,30 +220,80 @@ export function getQualifiedRouteComponent(value: RouteNode) {
 }
 
 /** @returns a function which provides a screen id that matches the dynamic route name in params. */
-export function createGetIdForRoute(route: Pick<RouteNode, 'dynamic' | 'route'>) {
-  if (!route.dynamic?.length) {
-    return undefined;
+export function createGetIdForRoute(
+  route: Pick<RouteNode, 'dynamic' | 'route' | 'contextKey' | 'children'>
+) {
+  const exclude = new Set<string>();
+  const include = new Map<string, DynamicConvention>();
+
+  if (route.dynamic) {
+    for (const segment of route.dynamic) {
+      include.set(segment.name, segment);
+    }
   }
-  return ({ params }: { params?: Record<string, any> }) => {
-    const getPreferredId = (segment: DynamicConvention) => {
-      // Params can be undefined when there are no params in the route.
-      const preferredId = params?.[segment.name];
-      // If the route has a dynamic segment, use the matching parameter
-      // as the screen id. This enables pushing a screen like `/[user]` multiple times
-      // when the user is different.
-      if (preferredId) {
-        if (!Array.isArray(preferredId)) {
-          return preferredId;
-        } else if (preferredId.length) {
-          // Deep dynamic routes will return as an array, so we'll join them to create a
-          // fully qualified string.
-          return preferredId.join('/');
+
+  /**
+   * Child routes IDs are a combination of their dynamic segments and the search parameters
+   * As search parameters can be anything, we build an exclude list of its parents dynamic segments.
+   **/
+  if (route.children?.length === 0) {
+    exclude.add('screen');
+    exclude.add('params');
```

Comment:
These are internal parameters added by React Navigation. We shouldn't include them in the ID